cpp/src/arrow/json/from_string.cc

```diff
@@ -55,14 +55,12 @@ namespace arrow {
 using internal::ParseValue;
 using util::Float16;
 
-namespace ipc {
-namespace internal {
 namespace json {
 
 using ::arrow::internal::checked_cast;
 using ::arrow::internal::checked_pointer_cast;
 
-namespace {
+namespace internal {
```

Comment:
Why not keep this in the anonymous namespace as it was?

Reply:
Yes, it would be better to rename IMHO.

Reply:
Closed by PR https://github.com/apache/arrow/pull/46697.